src/server.rs

```diff
@@ -1,21 +1,71 @@
 //! Faucet server implementation.
 
+use std::collections::HashSet;
+use std::net::{IpAddr, ToSocketAddrs as _};
+use std::str::FromStr as _;
+use std::sync::RwLock;
+use std::time::Duration;
+
 use actix_cors::Cors;
 use actix_web::http::{header, StatusCode};
-use actix_web::web::{get, post, Bytes};
-use actix_web::{App, HttpResponse, HttpServer, Responder};
-use eyre::Result;
+use actix_web::web::{get, post, Bytes, Data};
+use actix_web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use eyre::{eyre, Result};
 use tracing::{error, info};
 
-use crate::{active_requests, config, erc20_tokens, id, neon_token};
+use crate::{active_requests, config, erc20_tokens, id, neon_token, solana};
+
+type AirdropLimiter = Data<RwLock<neon_token::AirdropLimiter>>;
```

Comment:
Can we use a `tokio::sync::RwLock`? This would prevent blocking the thread.

Reply:
crap good catch. I will have to make everything `async fn` in order for that to work

Reply:
fixed - since using `tokio::spawn` for the cache clearing, no additional async fn were required